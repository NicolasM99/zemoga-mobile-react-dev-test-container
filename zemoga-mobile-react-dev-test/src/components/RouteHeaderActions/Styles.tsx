import { Platform, StyleSheet } from 'react-native';

export const routeHeaderStyles = StyleSheet.create({
  routeHeaderContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 20
  },
  routeHeaderTitle: {
    flex: 1,
    fontSize: 20,
    color: 'black'
  },
  routeHeaderActionsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingRight: Platform.OS === 'ios' ? 0 : 16
  },
  routeHeaderAction: {
    marginLeft: 24
  },
  routeHeaderGoBackBtnContainer: {
    marginRight: 16
  }
});
