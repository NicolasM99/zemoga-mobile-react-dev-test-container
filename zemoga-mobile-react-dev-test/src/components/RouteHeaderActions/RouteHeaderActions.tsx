import React, { FC } from 'react';

import FontAwesome from '@expo/vector-icons/FontAwesome';
import { TouchableOpacity, View } from 'react-native';

import { routeHeaderStyles } from './Styles';

export interface IRouteHeaderAction {
  ActionBtn: FC;
  name: string;
}

interface RouteHeaderActionProps {
  actions: Array<IRouteHeaderAction> | [];
}

const RouteHeaderActions: FC<RouteHeaderActionProps> = ({
  actions = []
}: RouteHeaderActionProps) => (
  <View style={routeHeaderStyles.routeHeaderActionsContainer}>
    {actions.map(({ ActionBtn, name }) => (
      <ActionBtn key={name} />
    ))}
  </View>
);

export default RouteHeaderActions;
