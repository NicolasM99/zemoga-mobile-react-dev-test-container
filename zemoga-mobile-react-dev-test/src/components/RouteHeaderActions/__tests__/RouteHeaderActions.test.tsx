import { render, screen } from '@testing-library/react-native';

import RouteHeaderActions from '../RouteHeaderActions';

describe('<RouteHeaderActions /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<RouteHeaderActions actions={[]} />);
    expect(screen).toMatchSnapshot();
  });
});
