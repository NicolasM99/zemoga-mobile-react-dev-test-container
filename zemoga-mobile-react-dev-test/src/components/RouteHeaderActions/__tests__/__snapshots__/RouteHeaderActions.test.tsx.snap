// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteHeaderActions /> unit tests Should mount correctly the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
/>
`;
